packages/driver/src/cross-origin/events/cookies.ts

```diff
@@ -0,0 +1,47 @@
+import type { AutomationCookie } from '@packages/server/lib/automation/cookies'
+import type { ICypress } from '../../cypress'
+
+// cross-origin cookies collected by the the proxy are sent down to the driver
+// via this event, so that they can be set via automation once the page has
+// loaded. it's necessary to wait until page load because Firefox with the
+// extension will hang the page load if we attempt to set the cookies via
+// automation before the page loads
+export const handleCrossOriginCookies = (Cypress: ICypress) => {
+  // multiple requests could set cookies while the page is loading, so we
+  // collect all cookies and only send set them via automation once after
+  // the page has loaded
+  let cookiesToSend: AutomationCookie[] = []
+  let waitingToSend = false
+
+  Cypress.on('cross:origin:cookies', (cookies: AutomationCookie[]) => {
+    Cypress.backend('cross:origin:cookies:received')
+
+    cookiesToSend = cookiesToSend.concat(cookies)
+
+    const onBeforeStabilityRelease = (stable: boolean) => {
+      if (!stable) return
```

Comment:
It's so weird that 'before:stability:release' can get called with false stability, i would have thought that this action would be called when stability is false but just before it is set to true.

Reply:
Good point. Changed it to behave like that: https://github.com/cypress-io/cypress/pull/23643/commits/f1441f0c59522b19d6b9a05863f852fc2bf2c7e1